agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/init/SecondEntryPoint.java

```diff
@@ -146,13 +144,33 @@ public void customize(AutoConfigurationCustomizer autoConfiguration) {
             telemetryClient.getMetricsBatchItemProcessor().trackAsync(telemetryItem);
           }
         };
-    HeartbeatExporter.start(
-        intervalSeconds, telemetryClient::populateDefaults, telemetryItemsConsumer);
+
+    if (telemetryClient.getConnectionString() != null) {
+      startupLogger.verbose("connection string is not null, start HeartbeatExporter");
+      // interval longer than 15 minutes is not allowed since we use this data for usage telemetry
+      long intervalSeconds =
+          Math.min(configuration.heartbeat.intervalSeconds, MINUTES.toSeconds(15));
+      HeartbeatExporter.start(
+          intervalSeconds, telemetryClient::populateDefaults, heartbeatTelemetryItemConsumer);
+    }
 
     TelemetryClient.setActive(telemetryClient);
 
+    if (configuration.preview.profiler.enabled && telemetryClient.getConnectionString() != null) {
+      try {
+        ProfilingInitializer.initialize(tempDir, configuration, telemetryClient);
```

Comment:
pass in
* `Configuration.ProfilerConfiguration profilerConfig`
* roleName
* roleInstance